Tests/UnitTests/Core/Seeding/PathSeederTest.cpp

```diff
@@ -0,0 +1,408 @@
+
+  std::vector<const Surface*> m_surfaces;
+
+  // Find the intersections along the path
+  // and return them in the order of the path
+  // length
+  std::vector<std::pair<GeometryIdentifier, Vector3>> operator()(
+      const GeometryContext& geoCtx, const Vector3& position,
+      const Vector3& direction, [[maybe_unused]] const ActsScalar& Pmag = 0,
+      [[maybe_unused]] const ActsScalar& Charge = 0) const {
+    std::vector<std::pair<GeometryIdentifier, Vector3>> sIntersections;
+    // Intersect the surfaces
+    for (auto& surface : m_surfaces) {
+      // Get the intersection
+      auto sMultiIntersection =
+          surface->intersect(geoCtx, position, direction,
+                             BoundaryTolerance::AbsoluteCartesian(tol, tol));
+
+      // Take the closest
+      auto closestForward = sMultiIntersection.closestForward();
+
+      // Store if the intersection is reachable
+      if (closestForward.status() == IntersectionStatus::reachable &&
+          closestForward.pathLength() > 0.0) {
+        sIntersections.push_back(
+            {closestForward.object()->geometryId(), closestForward.position()});
+        continue;
+      }
+    }
+    return sIntersections;
+  }
+};
+
+// Grid to store the source links for
+// the seeding fast lookup
+class SourceLinkGrid : public ISourceLinkGrid<Axis> {
+ public:
+  using eGrid = Grid<std::vector<SourceLink>, Axis, Axis>;
+
+  /// Lookup table collection
+  std::unordered_map<int, eGrid> m_lookupTables;
+
+  /// Surface accessor
+  SourceLinkSurfaceAccessor m_surfaceAccessor;
+
+  void initialize(const GeometryContext& geoCtx,
+                  std::vector<SourceLink> sourceLinks) override {
+    // Lookup table for each layer
+    std::unordered_map<int, eGrid> lookupTable;
+
+    // Construct a binned grid for each layer
+    for (int i : {14, 15, 16, 17}) {
+      Axis xAxis(-halfY, halfY, 50);
+      Axis yAxis(-halfZ, halfZ, 50);
+
+      eGrid grid(std::make_tuple(xAxis, yAxis));
+      lookupTable.insert({i, grid});
+    }
+    // Fill the grid with source links
+    for (auto& sl : sourceLinks) {
+      auto ssl = sl.get<detail::Test::TestSourceLink>();
+      auto id = ssl.m_geometryId;
+
+      // Grid works with global positions
+      Vector3 globalPos = m_surfaceAccessor(sl)->localToGlobal(
+          geoCtx, ssl.parameters, Vector3{0, 1, 0});
+
+      auto bin =
+          lookupTable.at(id.sensitive())
+              .localBinsFromPosition(Vector2(globalPos.y(), globalPos.z()));
+      lookupTable.at(id.sensitive()).atLocalBins(bin).push_back(sl);
+    }
+
+    m_lookupTables = lookupTable;
+  };
+
+  // Get the source link grid for a given geometry id
+  eGrid getSourceLinkTable(const GeometryIdentifier& geoId) const override {
+    return m_lookupTables.at(geoId.sensitive());
+  }
+};
+
+// A simple path width provider to set
+// the grid lookup boundaries around the
+// intersection point
+class PathWidthProvider {
+ public:
+  ActsScalar m_pathWidth = 0.1;
```

Comment:
Nitpick: you never set this to anything else, so this could be a simple function
```cpp
std::pair<ActsScalar, ActsScalar> getPathWidth() {
  return {0.1, 0.1};
}
```